### PR TITLE
Experimental dsl for displaying recyclable lists 

### PR DESCRIPTION
**Pull request by sgrekov**
This PR adds experimental dsl for build lists of data with much more convenient syntax then using regular recyclerView.

### CODE REVIEW

File: anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        v<RecyclerView>(configure.bind(InkrementalListScope))
+
+inline fun hList(crossinline r: () -> Unit) {
+    list {
+        layout(RecyclerLayoutType.Horizontal())
+        r.invoke()
+    }
+}
+
+inline fun vList(crossinline r: () -> Unit) {
```
Comment (r4zzz4k) [inline anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt]: You were probably going to add `InkrementalListScope.` receiver to `r` here, in `hList` and in `grid` builders :) You can replace `r.invoke()` with `r()`, which would pass the receiver automatically.

Comment (r4zzz4k) [reply anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt]: By the way, these seem quite useful, I'm thinking we should add same for LinearLayout. Though vLayout and hLayout would be not as obvious. Should we use `verticalList` + `horizontalList` here and `verticalLayout` + `horizontalLayout` there to keep stuff consistent?

File: anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+object InkrementalListSetter : Inkremental.AttributeSetter<Any> {
+    override fun set(v: View, name: String, value: Any?, prevValue: Any?): Boolean = when (name) {
+        "items" -> when {
+            v is RecyclerView && value is HolderAttr<*> -> {
+                val holder = value as HolderAttr<Any>
+                if (v.adapter == null) {
+                    v.adapter = RenderableRecyclerViewAdapter.withItems(holder.items) { index, letter ->
+                        value.r(index, letter)
```
Comment (r4zzz4k) [inline anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt]: Not that it matters much, but here you use `value.r`, and below `holder.r`. Both are fine, it's just a bit easier for reader to see same construct for both attributes.

**Review by r4zzz4k — APPROVED** [review]
Didn't try advanced stuff yet, but for basic usage seems pretty solid!
I guess we can always revisit this in following PRs, so I propose merging this as is and gather feedback.

File: anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt (hunk 2 above)
Comment (sgrekov) [reply anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt]: Yep, good catch!

File: anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt (hunk 1 above)
Comment (sgrekov) [reply anvil-recyclerview-v7/src/main/kotlin/dev/inkremental/dsl/androidx/recyclerview/InkrementalListSetter.kt]: Done!

**Comment by sgrekov** [comment]
> I guess we can always revisit this in following PRs, so I propose merging this as is and gather feedback.

Alright! Removing WIP then.

**Review by Mishkun — APPROVED** [review]

**Pull request merged**